src/roboptim-core-plugin-knitro.cc

```diff
@@ -250,8 +168,8 @@ namespace roboptim
     int result = 0;
 
     for (iterator_t it = this->problem ().constraints ().begin ();
-	 it != this->problem ().constraints ().end (); ++it)
-	result += boost::apply_visitor (ConstraintsSizeVisitor (), *it);
+	               it != this->problem ().constraints ().end (); ++it)
+        result += static_cast<int> ((*it)->outputSize ());
 
     return result;
   }
```

Comment:
There's actually a new method for this: [constraintsOutputSize](https://github.com/roboptim/roboptim-core/blob/master/include/roboptim/core/problem.hh#L276-L277)